dll/win32/kernel32/client/synch.c

```diff
@@ -446,13 +446,67 @@ CreateSemaphoreW(IN LPSECURITY_ATTRIBUTES lpSemaphoreAttributes  OPTIONAL,
                  IN LONG lMaximumCount,
                  IN LPCWSTR lpName  OPTIONAL)
 {
-    CreateNtObjectFromWin32Api(Semaphore, Semaphore, SEMAPHORE,
+    CreateNtObjectFromWin32Api(Semaphore, Semaphore, SEMAPHORE_ALL_ACCESS,
                                lpSemaphoreAttributes,
                                lpName,
                                lInitialCount,
                                lMaximumCount);
 }
 
+HANDLE
+WINAPI
+DECLSPEC_HOTPATCH
+CreateSemaphoreExW(IN LPSECURITY_ATTRIBUTES lpSemaphoreAttributes  OPTIONAL,
+                   IN LONG lInitialCount,
+                   IN LONG lMaximumCount,
+                   IN LPCWSTR lpName  OPTIONAL,
+                   IN DWORD dwFlags OPTIONAL,
```

Comment:
It seems you're ignoring dwFlags? Seems like there should be some sort of debug print in case it's nonzero.

Reply:
@ThFabba yes, as i understood it's fine to ignore him. In MSDN this param have this description - "This parameter is reserved and must be 0.". 

Add debug print for non-zero input?